### ISSUE TITLE
DeepOBS: A Deep Learning Optimizer Benchmark Suite

### ISSUE THREAD

**Issue by sile** [sile/readings#6]
URL
-----

- https://arxiv.org/abs/1903.05499
- https://github.com/fsschneider/DeepOBS
- https://deepobs.readthedocs.io/en/latest/

Abstract
----------

> Because the choice and tuning of the optimizer affects the speed, and ultimately the performance of deep learning, there is significant past and recent research in this area. Yet, perhaps surprisingly, there is no generally agreed-upon protocol for the quantitative and reproducible evaluation of optimization strategies for deep learning. We suggest routines and benchmarks for stochastic optimization, with special focus on the unique aspects of deep learning, such as stochasticity, tunability and generalization. As the primary contribution, we present DeepOBS, a Python package of deep learning optimization benchmarks. The package addresses key challenges in the quantitative assessment of stochastic optimizers, and automates most steps of benchmarking. The library includes a wide and extensible set of ready-to-use realistic optimization problems, such as training Residual Networks for image classification on ImageNet or character-level language prediction models, as well as popular classics like MNIST and CIFAR-10. The package also provides realistic baseline results for the most popular optimizers on these test problems, ensuring a fair comparison to the competition when benchmarking new optimizers, and without having to run costly experiments. It comes with output back-ends that directly produce LaTeX code for inclusion in academic publications. It supports TensorFlow and is available open source. 

Table of Contents
--------------------

- (1) Introduction
  - (1-1) Related Works
- (2) Benchmarking Deep Learning Optimizers
  - (2-1) Stochasticity
  - (2-2) Choice of Performance Metric
  - (2-3) Measuring Speed
  - (2-4) Hyperparameter Tuning
- (3) Benchmark Suite Overview
  - (3-1) Data Loading
  - (3-2) Models
  - (3-3) Runners
  - (3-4) Baselines
  - (3-5) Estimate Runtime
  - (3-6) Visualizations
- (4) Insights from The Baselines
- (5) Conclusion

BibTex
--------

```
@article{DBLP:journals/corr/abs-1903-05499,
  author    = {Frank Schneider and
               Lukas Balles and
               Philipp Hennig},
  title     = {DeepOBS: {A} Deep Learning Optimizer Benchmark Suite},
  journal   = {CoRR},
  volume    = {abs/1903.05499},
  year      = {2019},
  url       = {http://arxiv.org/abs/1903.05499},
  archivePrefix = {arXiv},
  eprint    = {1903.05499},
  timestamp = {Sun, 31 Mar 2019 19:01:24 +0200},
  biburl    = {https://dblp.org/rec/bib/journals/corr/abs-1903-05499},
  bibsource = {dblp computer science bibliography, https://dblp.org}
}
```

**Comment by sile** [comment]
> we strongly recommend reporting both loss and accuracy, for both training and test set, when demonstrating a new optimizer as there is no obvious way those four learning curves are connected in general.

**Comment by sile** [comment]
>  For hyperparameter tuning, we use test accuracy or, if that is not available, test loss, as the criteria.
